src/contexts/User-Context/user-context.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useContext, useReducer } from "react";
+import { Navigate, useLocation } from "react-router-dom";
+import { callAPI } from "../../app-utils";
+import { initialUserState, userReducer } from "./user-utils";
+
+const AuthContext = createContext(initialUserState);
+
+const AuthProvider = ({ children }) => {
+
+  const [userState, userDispatch] = useReducer(userReducer, initialUserState);
+
+  const loginUser = async (credentials) => {
+    try {
+      const { data } = await callAPI("POST", "/api/auth/login", credentials);
+      console.log(data);
+    } catch (error) {
+      console.log(error);
```

Comment:
Here  *console.log* can be removed.

And, In *catch block* you can use **console.error** so that it appears as error in console rather than a standard output (ie. you get using *console.log*)

Useful in dealing with **errors** specifically.